### PR TITLE
feat: First pass downloadMany

### PR DESCRIPTION
**Pull request by sydney-munro**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-storage/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

If you write sample code, please follow the [samples format](
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md).

### CODE REVIEW

File: google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITTransferManagerTest.java [hunk 1]
```diff
@@ -79,5 +109,14 @@ public void uploadFiles() throws IOException {
   }
 
   @Test
-  public void downloadBlobs() {}
+  public void downloadBlobs() throws IOException {
+    TransferManagerConfig config =
+        TransferManagerConfig.newBuilder().setAllowChunking(false).setMaxWorkers(1).build();
```
Comment (danielduhh) [inline google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITTransferManagerTest.java:114]: Isn't setAllowChunking false by default?

File: google-cloud-storage/src/main/java/com/google/cloud/storage/transfermanager/DownloadCallable.java [hunk 2]
```diff
@@ -16,4 +16,64 @@
...
+      BlobInfo originalBlob,
+      ParallelDownloadConfig parallelDownloadConfig) {
+    this.transferManagerConfig = transferManagerConfig;
+    this.originalBlob = originalBlob;
+    this.parallelDownloadConfig = parallelDownloadConfig;
+  }
+
+  @Override
+  public DownloadResult call() throws Exception {
+    // TODO: Check for chunking
```
Comment (danielduhh) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/transfermanager/DownloadCallable.java:47]: What's the plan here? To check `setAllowChunking`, and decide whether to download byte stream ?

File: google-cloud-storage/src/main/java/com/google/cloud/storage/transfermanager/TransferManagerImpl.java [hunk 3]
```diff
@@ -56,6 +56,14 @@ public TransferManagerImpl(TransferManagerConfig transferManagerConfig) {
 
   @Override
   public @NonNull DownloadJob downloadBlobs(List<BlobInfo> blobs, ParallelDownloadConfig opts) {
```
Comment (danielduhh) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/transfermanager/TransferManagerImpl.java:58]: Docs?

**Review by danielduhh — COMMENTED** [review]
Looks solid just had a couple questions & request for method docs

**Comment by ddelgrosso1** [comment]
Overall the implementation looks good to me. I will leave it to the Java experts to look more closely at the specifics.

File: google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITTransferManagerTest.java (hunk 1 above)
Comment (sydney-munro) [reply google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITTransferManagerTest.java:114]: yes it is, can remove that.

File: google-cloud-storage/src/main/java/com/google/cloud/storage/transfermanager/TransferManagerImpl.java (hunk 3 above)
Comment (sydney-munro) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/transfermanager/TransferManagerImpl.java:58]: docs will be a followup, just getting first passes out first.

**Review by JesseLovelace — APPROVED** [review]

**Pull request merged**